idseq_dag/util/server.py

```diff
@@ -173,6 +185,42 @@ def wait_for_server_ip(service_name,
         # if we had to wait here, that counts toward the rate limit delay
         result = wait_for_server_ip_work(service_name, key_path,
                                          remote_username, environment,
-                                         max_concurrent, chunk_id)
+                                         max_concurrent, chunk_id,
+                                         max_interval_between_describe_instances,
+                                         draining_tag)
         return result
 
+
+def build_job_tag(job_tag_prefix, chunk_id):
+    batch_job_id = os.environ.get('AWS_BATCH_JOB_ID', 'local')
+    job_tag_key = f"{job_tag_prefix}{batch_job_id}_chunk{chunk_id}"
+    job_tag_value = int(time.time())
+    return job_tag_key, job_tag_value
+
+
+def create_tag(instance_iD, tag_key, tag_value):
+    command.execute(f"aws ec2 create-tags --resources {instance_iD} --tags Key={tag_key},Value={tag_value}")
+
+def delete_tag(instance_iD, tag_key):
+    command.execute(f"aws ec2 delete-tags --resources {instance_iD} --tags Key={tag_key}")
+
+
+@contextmanager
+def ASGInstance(service, key_path, remote_username, environment, max_concurrent, chunk_id,
+                max_interval_between_describe_instances=900,
+                job_tag_prefix="RunningIDseqBatchJob_",
+                job_tag_refresh_seconds=600,
+                draining_tag="draining"):
+    instance_ip, instance_iD = wait_for_server_ip(service, key_path, remote_username, environment, max_concurrent, chunk_id,
+        max_interval_between_describe_instances, draining_tag)
+    log.write(f"starting alignment for chunk {chunk_id} on {service} server {instance_ip}")
+    job_tag_key, job_tag_value = build_job_tag(job_tag_prefix, chunk_id)
```

Comment:
It seems you are capturing the time into the tag value ONCE at the time you launch the periodic thread;  but I would have expected the desired behavior to be that you capture and update the time in the tag on every invocation of the target by the periodic thread.

Reply:
good catch, thanks!